sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -0,0 +1,256 @@
+                                                                                       String requestId,
+                                                                                       ServiceCallBack<ContainersListBlobFlatSegmentResponse> callback) {
+        final String resType = "container";
+        final String comp = "list";
+        if (callback != null) {
+            executeCall(service.listBlobFlatSegment(containerName,
+                prefix,
+                pageId,
+                maxResults,
+                this.serializerAdapter.serializeList(include, SerializerAdapter.CollectionFormat.CSV),
+                timeout,
+                XMS_VERSION,
+                requestId,
+                resType,
+                comp
+                /*context*/), new Callback<ResponseBody>() {
+                @Override
+                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                    if (response.isSuccessful()) {
+                        if (response.code() == 200) {
+                            ListBlobsFlatSegmentResponse typedContent = deserializeContent(response.body(),
+                                ListBlobsFlatSegmentResponse.class);
+                            ContainerListBlobFlatSegmentHeaders typedHeader = deserializeHeaders(response.headers(),
+                                ContainerListBlobFlatSegmentHeaders.class);
+                            callback.onResponse(new ContainersListBlobFlatSegmentResponse(response.raw().request(),
+                                response.code(),
+                                response.headers(),
+                                typedContent,
+                                typedHeader));
+                        } else {
+                            String strContent = readAsString(response.body());
+                            callback.onFailure(new BlobStorageException(strContent, response.raw()));
+                        }
+                    } else {
+                        String strContent = readAsString(response.errorBody());
+                        callback.onFailure(new BlobStorageException(strContent, response.raw()));
+                    }
+                }
+
+                @Override
+                public void onFailure(Call<ResponseBody> call, Throwable t) {
+                    callback.onFailure(t);
+                }
+            });
+            return null;
+        } else {
+            Response<ResponseBody> response = executeCall(service.listBlobFlatSegment(containerName,
+                prefix,
+                pageId,
+                maxResults,
+                this.serializerAdapter.serializeList(include, SerializerAdapter.CollectionFormat.CSV),
+                timeout,
+                XMS_VERSION,
+                requestId,
+                resType,
+                comp
+                /*context*/));
```

Comment:
Info: commented context will be removed, we haven't encountered need for pipeline context so far in mobile, so putting it in back log.